### PR TITLE
breaking: bump terser

### PR DESCRIPTION
**Pull request by lifeart**
because of differences in terser optimization logic (from v3 to v5), snapshots need to be changed, to simplify support, tests migrated to jest

dropped node 6, 8 support, 
new minimal version is 10

### CODE REVIEW

**Comment by lifeart** [comment]
I'm getting different stats in my local and in CI

image

and this is why CI is failing..

**Comment by rwjblue** [comment]
Generally speaking, all of these things should be separate PRs:

* snapshots need to be changed, to simplify support, tests migrated to jest
* dropped node 6, 8 support
* update terser

I don't have commit/release here, so I'll leave it to @stefanpenner / @simonihmig on how to proceed.

**Comment by simonihmig** [comment]
@lifeart thanks for working on this!

> Generally speaking, all of these things should be separate PRs:

Generally speaking, I would agree! 😝 Alex, would you mind splitting this up? I guess bumping terser (with the changed test date) would come last, so you can use jest snapshots without having to manually update the text fixtures. At least that's why you wanted to introduce jest here mainly, right?

> I'm getting different stats in my local and in CI

My best *guess* would be that it could be related to line endings, as the numbers are quite close, and only the "raw" numbers diverge. I guess you are running this on Windows, right?

**Comment by lifeart** [comment]
@rwjblue, @simonihmig thank you for feedback, yeah I think I will try to split it to: jest, node, terser Mr’s

And yes, I’m running tests and snapshots on windows

File: lib/summarize.js [hunk 1]
```diff
@@ -8,12 +8,11 @@ const pify = require('pify');
 
 function uglify(content) {
   return new Promise((resolve, reject) => {
```
Comment (stefanpenner) [inline lib/summarize.js:10]: I believe the following is a more concise (with better stack-traces/debugging approach)

```js
async function uglify(content) {
  const { code } = await Tester.minify(content);
  return code;
}
```

**Review by stefanpenner — CHANGES_REQUESTED** [review]

**Comment by lifeart** [comment]
@stefanpenner, @simonihmig 

1.) https://github.com/stefanpenner/broccoli-concat-analyser/pull/53
2.) https://github.com/stefanpenner/broccoli-concat-analyser/pull/54
3.) https://github.com/stefanpenner/broccoli-concat-analyser/pull/55

**Pull request closed without merging**